python/paddle/utils/deprecated.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+decorator to deprecate a function or class
+"""
+
+import warnings
+import functools
+import paddle
+
+
+def deprecated(update_to="", since="", reason=""):
+    """Decorate a function to signify its deprecation.
+
+       This function wraps a method that will soon be removed and does two things:
+           - The docstring of the API will be modified to include a notice
+             about deprecation."
+           - Raises a :class:`~exceptions.DeprecatedWarning` when old API is called.
+       Args:
+           since(str): The version at which the decorated method is considered deprecated.
+           update_to(str): The new API users should use.
+           reason(str): The reason why the API is deprecated.
+       Returns:
+           decorator: decorated function or class.
+    """
+
+    def _compare_version(v1, v2):
+        """compare two versions."""
+
+        _v1 = [int(i) for i in v1.split(".")] + [0] * (4 - len(v1))
+        _v2 = [int(i) for i in v2.split(".")] + [0] * (4 - len(v2))
+        for i in range(4):
```

Comment:
you only need `v1>= v2`, or `!(v1 < v2)`, so simple true and false should suffice

Reply:
this function should be inlined, as it is only 2 line now...

Reply:
done